dle/search/services.py

```diff
@@ -30,55 +32,94 @@ def build_match_query(search_query: str) -> str:
+    if search_request.select_section:
+      select_clause += """ AND LOWER(section_name) = %(section_name)s"""
+  
+
+    return "WITH cte AS (" + select_clause + ")"
+
 def process_search(search_request: SearchRequest) -> List[DrugLabel]:
     search_filter_mapping = {
-        "select_section": "section_name",
         "select_agency": "source",
         "manufacturer_input": "marketer",
         "generic_name_input": "generic_name",
         "brand_name_input": "product_name",
     }
     search_request_dict = search_request._asdict()
-    search_query = search_request.search_text.lower()
-    raw_sql = """
+    with_clause = build_with_clause(search_request=search_request)
+    select_sql = """
         SELECT
-            DISTINCT dl.id,
-            dl.source,
-            dl.product_name,
-            dl.generic_name,
-            dl.version_date,
-            dl.source_product_number,
-            dl.marketer,
-            dl.link
+        dl.id,
+        dl.source,
+        dl.product_name,
+        dl.generic_name,
+        dl.version_date,
+        dl.source_product_number,
+        pcte.section_text as raw_text,
+        dl.marketer,
+        dl.link
         FROM
-            data_druglabel as dl
-        JOIN data_labelproduct as lp ON dl.id = lp.drug_label_id
-        JOIN data_productsection as ps ON lp.id = ps.label_product_id
+        (
+            SELECT
+            label_product_id,
+            id as section_id,
+            score,
+            ROW_NUMBER() OVER(
+                PARTITION BY label_product_id
+                ORDER BY
+                score DESC
+            ) as score_rank
+            FROM
+            cte
+            GROUP BY
+            1,
+            2
+        ) as cte2
+        JOIN data_labelproduct AS lp ON cte2.label_product_id = lp.id
+        JOIN data_productsection AS pcte ON cte2.section_id = pcte.id
+        JOIN data_druglabel AS dl ON lp.drug_label_id = dl.id
         WHERE
-            1=1
+        cte2.score_rank = 1
         """
 
-    # exact match query
-    raw_sql += build_match_query(search_query)
+    sql_params = {"search_query": search_request.search_text}
+    if search_request.select_section:
+        sql_params["section_name"] =  search_request.select_section
 
-    sql_params = {"search_query": search_query}
     for k, v in search_request_dict.items():
         if v and k in search_filter_mapping:
             param_key = search_filter_mapping[k]
             if k == "select_section" and not v:
                 continue
             sql_params[param_key] = v
             additional_filter = f"AND LOWER({param_key}) = %({param_key})s "
-            raw_sql += additional_filter
-    raw_sql += "LIMIT 30" #can remove this once we're done testing
-    print(raw_sql % sql_params)
-    return [d for d in DrugLabel.objects.raw(raw_sql, params=sql_params)]
+            select_sql += additional_filter
+    
+    order_limit_clause = """
+        ORDER BY cte2.score DESC
```

Comment:
I'm wondering if this order by is even necessary, since any section returned from the match query might be able to be highlighted, wouldn't be always guaranteed though...